extensions/objects/templates/math-expression-content-editor.directive.ts

```diff
@@ -35,15 +39,58 @@ angular.module('oppia').directive('mathExpressionContentEditor', [
       controllerAs: '$ctrl',
       controller: ['$scope', function($scope) {
         var ctrl = this;
+        var convertLatexStringToSvg = function(inputLatexString) {
+          var emptyDiv = document.createElement('div');
+          var outputElement = angular.element(emptyDiv);
+          var $script = angular.element(
+            '<script type="math/tex">'
+          ).html(inputLatexString === undefined ? '' : inputLatexString);
+          outputElement.html('');
+          outputElement.append($script);
+          MathJax.Hub.Queue(['Typeset', MathJax.Hub, outputElement[0]]);
+          MathJax.Hub.Queue(function() {
+            ctrl.svgString = (
+              outputElement[0].getElementsByTagName('svg')[0].outerHTML);
+          });
+        };
+
+        var processAndSaveSvg = function() {
+          var cleanedSvgString = (
+            ImageUploadHelperService.cleanMathExpressionSvgString(
+              ctrl.svgString));
+          var dimensions = (
+            ImageUploadHelperService.
+              extractDimensionsFromMathExpressionSvgString(cleanedSvgString));
+          var fileName = (
+            ImageUploadHelperService.generateMathExpressionImageFilename(
+              dimensions.height, dimensions.width, dimensions.verticalPadding));
+          var dataURI = 'data:image/svg+xml;base64,' + btoa(cleanedSvgString);
+          var invalidTagsAndAttributes = (
+            ImageUploadHelperService.getInvalidSvgTagsAndAttrs(
+              dataURI));
+          var tags = invalidTagsAndAttributes.tags;
+          var attrs = invalidTagsAndAttributes.attrs;
+          if (tags.length === 0 && attrs.length === 0) {
+            ctrl.value.svgFile = dataURI;
+            ctrl.value.svg_filename = fileName;
+          } else {
+            ctrl.value.raw_latex = '';
+            ctrl.value.svg_filename = '';
+            AlertsService.addWarning('SVG failed validation.');
```

Comment:
will this be shown after closing the modal?

Reply:
Yes if the validation fails, the modal is closed and this warning is shown.